app/services/curation_concerns/working_directory.rb

```diff
@@ -48,7 +48,7 @@ def copy_stream_to_working_directory(id, name, stream)
         end
 
         def full_filename(id, original_name)
-          pair = id.scan(/..?/).first(4)
+          pair = id.scan(/..?/)
```

Comment:
This has the potential to create very deep directory structures, doesn't it? To avoid this, might it be better to add the original id as the last leaf node?